clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -0,0 +1,1021 @@
+     * @param <T>                               The KafkaFuture result type.
+     */
+    private static <T> void completeAllExceptionally(Collection<KafkaFuture<T>> futures, Throwable exc) {
+        for (KafkaFuture<?> future : futures) {
+            future.completeExceptionally(exc);
+        }
+    }
+
+    /**
+     * Get the current time remaining before a deadline as an integer.
+     *
+     * @param now                               The current time in milliseconds.
+     * @param deadlineMs                        The deadline time in milliseconds.
+     * @return                                  The time delta in milliseconds.
+     */
+    static int getTimeoutMsRemainingAsInt(long now, long deadlineMs) {
+        long deltaMs = deadlineMs - now;
+        if (deltaMs > Integer.MAX_VALUE)
+            deltaMs = Integer.MAX_VALUE;
+        else if (deltaMs < Integer.MIN_VALUE)
+            deltaMs = Integer.MIN_VALUE;
+        return (int) deltaMs;
+    }
+
+    /**
+     * Generate the client id based on the configuration.
+     *
+     * @param config                            The configuration
+     *
+     * @return                                  The client id
+     */
+    static String generateClientId(AdminClientConfig config) {
+        String clientId = config.getString(AdminClientConfig.CLIENT_ID_CONFIG);
+        if (clientId.length() > 0)
+            return clientId;
+        return "adminclient-" + ADMIN_CLIENT_ID_SEQUENCE.getAndIncrement();
+    }
+
+    /**
+     * Get the deadline for a particular call.
+     *
+     * @param optionTimeoutMs                   The timeout option given by the user.
+     *
+     * @return                                  The deadline in milliseconds.
+     */
+    private long getDeadlineMs(Integer optionTimeoutMs) {
+        if (optionTimeoutMs != null)
+            return time.milliseconds() + Math.min(0, optionTimeoutMs);
+        // If optionTimeoutMs is less than 0, we use the default timeout.
+        return time.milliseconds() + defaultTimeoutMs;
+    }
+
+    /**
+     * Pretty-print an exception.
+     *
+     * @param throwable                         The exception.
+     *
+     * @return                                  A compact human-readable string.
+     */
+    static String prettyPrintException(Throwable throwable) {
+        if (throwable == null)
+            return "Null exception.";
+        if (throwable.getMessage() != null) {
+            return throwable.getClass().getSimpleName() + ": " + throwable.getMessage();
+        }
+        return throwable.getClass().getSimpleName();
+    }
+
+    /**
+     * Close an AutoCloseable, suppressing any exceptions.
+     *
+     * @param what                              Human-readable string explaining what we are closing.
+     * @param closeable                         What we are closing, or null to ignore.
+     */
+    static void closeQuietly(String what, AutoCloseable closeable) {
```

Comment:
Why not use `Utils.closeQuietly`? We can change the signature to take an `AutoCloseable` instead of `Closeable`, if that's the issue.